### PR TITLE
Fix RenderEditable.computeMaxIntrinsicWidth error return.

### LINKED ISSUE #82277: when call RenderEditable.computeMaxIntrinsicWidth, which is one pixel less than the actual width.
code: 

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  static const String _title = 'Flutter Code Sample';

  @override
  Widget build(BuildContext context) {
    var data = 'test test test test test test test test test test';
    return MaterialApp(
        title: _title,
        home: Scaffold(
          body: Column(
            children: [
              _TestWidget(child: Text(data),),
              _TestWidget(child: SelectableText(data),),
            ],
          ),
        ),
      );
  }
}

class _TestWidget extends StatelessWidget {
  const _TestWidget({
    Key? key,
    required this.child,
  }) : super(key: key);

  final Widget child;

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.all(32),
      child: IntrinsicWidth(
        child: Column(
          mainAxisSize: MainAxisSize.min,
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            Container(
              color: Colors.blue,
              padding: const EdgeInsets.all(8.0),
              width: 245,
            ),
            Wrap(
              alignment: WrapAlignment.spaceBetween,
              runAlignment: WrapAlignment.end,
              // spacing: 8,
              children: [
                child,
                Text('date time')
              ],
            ),
          ],
        ),
      ),
    );
  }
}
```

image

The SelectableText widget actual width is 292, but RenderEditable.computeMaxIntrinsicWidth return 291.

### PR DESCRIPTION
**Pull request by YeungKC**
close #82277

Because _kCaretGap is not calculated in RenderEditable.computeMaxIntrinsicWidth, an incorrect with may be returned.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/rendering/editable.dart [hunk 1]
```diff
@@ -3123,7 +3123,7 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
   @override
   double computeMaxIntrinsicWidth(double height) {
     _layoutText(maxWidth: double.infinity);
-    return _textPainter.maxIntrinsicWidth + cursorWidth;
+    return _textPainter.maxIntrinsicWidth + _caretMargin;
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/rendering/editable.dart:3126]: Should we add this to `computeMinIntrinsicWidth` too?

**Review by LongCatIsLooong — COMMENTED** [review]
Thanks for the fix! Could you add a test to verify that if you layout a `RenderEditable` with the exact width `getMaxIntrinsicWidth` gave you, the render editable does not have a second line?

**Comment by YeungKC** [comment]
I don't think it's necessary, and in fact the real problem is not the wrong line break.

File: packages/flutter/lib/src/rendering/editable.dart (hunk 1 above)
Comment (YeungKC) [reply packages/flutter/lib/src/rendering/editable.dart:3126]: About this I am not very clear, @justinmc how to think about it?

**Comment by LongCatIsLooong** [comment]
`getMaxIntrinsicWidth`'s documentation (https://master-api.flutter.dev/flutter/rendering/RenderBox/getMaxIntrinsicWidth.html) states:
> Returns the smallest width beyond which increasing the width never decreases the preferred height. The preferred height is the value that would be returned by getMinIntrinsicHeight for that width.

That means if you layout the `RenderEditable` with the width given by `getMaxIntrinsicWidth` it should never soft break right?

**Comment by YeungKC** [comment]
It does look like it needs to be increased, but I found at https://github.com/flutter/flutter/commit/9161ec42cb5cb9f070b177833aff0c6175e575a1 that @justinmc  doesn't have the computeMinIntrinsicWidth is increased, so I'm a bit unsure.

**Comment by LongCatIsLooong** [comment]
Yeah looking at the implementation of `TextPainter.minIntrinsicWidth`, if the `maxLines` parameter is 1 it returns `maxIntrinsicWidth`. So it should probably need to take the width of the cursor into account as well. It's totally fine to leave it as-is for now, I can open a new issue for it.

**Comment by YeungKC** [comment]
Awesome, thanks for the explanation.

So we should be able to merge this pr now?

**Review by LongCatIsLooong — APPROVED** [review]

**Pull request merged**